MapboxNavigation/LanesView.swift

```diff
@@ -66,31 +66,33 @@ open class LanesView: UIView {
         separatorView.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
     }
     
-    func update(for currentLegProgress: RouteLegProgress) {
-        guard let step = currentLegProgress.upComingStep else { return }
-        guard !currentLegProgress.userHasArrivedAtWaypoint else { return }
-        let durationRemaining = currentLegProgress.currentStepProgress.durationRemaining
-        
+    func update(for currentStepProgress: RouteStepProgress) {
         clearLaneViews()
         
-        if let allLanes = step.intersections?.first?.approachLanes,
-            let usableLanes = step.intersections?.first?.usableApproachLanes,
-            durationRemaining < RouteControllerMediumAlertInterval {
-            
-            for (i, lane) in allLanes.enumerated() {
-                let laneView = laneArrowView()
-                laneView.lane = lane
-                laneView.maneuverDirection = step.maneuverDirection
-                laneView.isValid = usableLanes.contains(i as Int)
-                stackView.addArrangedSubview(laneView)
-            }
+        let distanceTraveled = currentStepProgress.distanceTraveled
+        guard let visualInstruction = currentStepProgress.currentVisualInstruction,
+              let tertiaryInstruction = visualInstruction.tertiaryInstruction,
+                  distanceTraveled <= visualInstruction.distanceAlongStep,
+                  tertiaryInstruction.containsLaneIndications
+                else {
+                    hide()
+                    return
         }
         
-        if stackView.arrangedSubviews.count > 0 {
-            show()
-        } else {
-            hide()
+        let laneIndications: [LaneIndicationComponent]? = visualInstruction.tertiaryInstruction?.components.compactMap({ component in
+            guard let lane = component as? LaneIndicationComponent else { return nil }
+            return lane
+        })
+        
+        guard let lanes = laneIndications, !lanes.isEmpty
+            else {
```

Comment:
nit, move this up a line.

Reply:
@bsudekum what do you mean? This was intentional so enhance readability.

Reply:
@vincethecoder Move the `else {` one line up from line 88 to 87.